### DISCUSSION TITLE
Upgrading Next.js, what do the carats do?

### DISCUSSION THREAD

**Discussion by tr1s** [vercel/next.js#12808] [Help] [answered]
On the [Next.js 9.4 blog post](https://nextjs.org/blog/next-9-4) it didn't give upgrade instructions but the [Next 9.3.5 blog post](https://nextjs.org/blog/next-9-3) did. I normally would have just `npm install next` but on the 9.3 blog it said to use `@latest`.

Doesn't running `npm i` always install the latest, how come we need the @latest flag at the end? and what do the little `^` carets do on my new `package.json`?

old:

```
  "dependencies": {
    "next": "9.3.5",
    "react": "16.13.1",
    "react-dom": "16.13.1",
  }
```

new with carets:

```
  "dependencies": {
    "next": "^9.4.0",
    "react": "^16.13.1",
    "react-dom": "^16.13.1",
  }
```

just curious, thanks.

**Comment by jamesmosier** [comment, marked as answer]
The carets in `package.json` will update you to all future minor/patch versions ([this Stack Overflow question answers it actually](https://stackoverflow.com/questions/22343224/whats-the-difference-between-tilde-and-caret-in-package-json)). So if you had `"next": "^9.4.0"` in a new project, ran `npm install` and the latest version of Next available was `9.4.1`, then it would install that `9.4.1` version for you.

After the initial installation, `package-lock.json` files will lock to whatever version you installed originally when the dependency was first installed (unless you manually increment the version yourself).

**Reply by tr1s** [reply]
I see. I get it, thanks!
